contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2905,6 +2905,7 @@ handleBatchLevelStatement(TSqlParser::Batch_level_statementContext *ctx, tsqlSel
 	result->body = list_make1(init);
 	// create PLtsql_stmt_execsql to wrap all query string
 	PLtsql_stmt_execsql *execsql = (PLtsql_stmt_execsql *) makeSQL(ctx);
+	execsql->original_query = (makeTsqlExpr(ctx, false))->query;
```

Comment:
Inside `makeTsqlExpr()`, we use `palloc()` to create the query string. When we use `palloc()`, the memory is allocated in the `CurrentMemoryContext`. In this PR, we don't use any deep copy and just passing `char *` from ANTLR to `pltsql_store_view_definition` hook. If the life cycle of `CurrentMemoryContext` is different between here and `pltsql_store_view_definition`, it could be problematic.

For more details, please refer https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/BABEL_3_X_DEV__PG_15_X/src/backend/utils/mmgr/README

Reply:
Used a duplicate string to avoid any problematic situation in future.